packages/sdk/src/client.ts

```diff
@@ -736,6 +721,151 @@ export function SoundClient({
     return network.chainId
   }
 
+  // Addresses with MINTER_ROLE for a given edition
+  async function _editionRegisteredMintersFromChain({
+    editionAddress,
+    fromBlockOrBlockHash,
+  }: {
+    editionAddress: string
+    fromBlockOrBlockHash: BlockOrBlockHash | undefined
+  }): Promise<string[]> {
+    const { signerOrProvider } = await _requireSignerOrProvider()
+
+    const editionContract = SoundEditionV1_1__factory.connect(editionAddress, signerOrProvider)
+    // Get the addresses with MINTER_ROLE
+    const minterRole = await editionContract.MINTER_ROLE()
+    const filter = editionContract.filters.RolesUpdated(undefined, minterRole)
+
+    const roleEvents = await editionContract.queryFilter(filter, fromBlockOrBlockHash)
+    const candidateMinters = roleEvents.map((event) => event.args.user)
+
+    // Check supportsInterface() to verify each address is a minter
+    const minters = await Promise.all(
+      candidateMinters.map(async (minterAddress) => {
+        const minterContract = IMinterModule__factory.connect(minterAddress, signerOrProvider)
+
+        try {
+          const isMinter = await minterContract.supportsInterface(interfaceIds.IMinterModule)
+          return isMinter ? minterAddress : null
+        } catch (err) {
+          onError(err)
+          return null
+        }
+      }),
+    )
+    // This list may contain duplicates if MINTER_ROLE was granted multiple times
+    const allMinters = minters.reduce((acc, minter) => {
+      if (minter) acc.add(minter)
+      return acc
+    }, new Set<string>())
+
+    return Array.from(allMinters)
+  }
+
+  // Addresses with MINTER_ROLE for a given edition
+  async function _editionRegisteredMintersFromSubgraph({
+    editionAddress,
+  }: {
+    editionAddress: string
+  }): Promise<string[] | null> {
+    const soundSubgraph = client.soundSubgraph
+    if (soundSubgraph) {
```

Comment:
Mind if we return early instead of nesting?

```
if (!subgraph) return null;
```

Reply:
https://github.com/soundxyz/sdk/pull/129/commits/18340f619cc4bae53b8e073f5c886e6a0a1329a5